src/main/java/com/microsoft/azure/spring/data/documentdb/repository/query/DocumentDbQueryCreator.java

```diff
@@ -7,23 +7,57 @@
 
 import com.microsoft.azure.spring.data.documentdb.core.mapping.DocumentDbPersistentProperty;
 import com.microsoft.azure.spring.data.documentdb.core.query.Criteria;
+import com.microsoft.azure.spring.data.documentdb.core.query.Criteria.CriteriaType;
 import com.microsoft.azure.spring.data.documentdb.core.query.Query;
-import org.apache.commons.lang3.NotImplementedException;
+import org.javatuples.Pair;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.data.domain.Sort;
-import org.springframework.data.mapping.PropertyPath;
 import org.springframework.data.mapping.context.MappingContext;
-import org.springframework.data.mapping.context.PersistentPropertyPath;
 import org.springframework.data.repository.query.parser.AbstractQueryCreator;
 import org.springframework.data.repository.query.parser.Part;
 import org.springframework.data.repository.query.parser.PartTree;
 
 import java.util.Iterator;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 
 public class DocumentDbQueryCreator extends AbstractQueryCreator<Query, Criteria> {
 
+    private static final Logger logger = LoggerFactory.getLogger(DocumentDbQueryCreator.class);
+
+    private static final Map<Part.Type, Pair<CriteriaType, Boolean>> criteriaLookup;
+    static {
+        final Map<Part.Type, Pair<CriteriaType, Boolean>> init = new HashMap<>();
+        init.put(Part.Type.BEFORE, new Pair<>(CriteriaType.IS_LESS_THAN, Boolean.FALSE));
+        init.put(Part.Type.AFTER, new Pair<>(CriteriaType.IS_LESS_THAN_OR_EQUAL, Boolean.TRUE));
+        init.put(Part.Type.GREATER_THAN, new Pair<>(CriteriaType.IS_GREATER_THAN_OR_EQUAL, Boolean.FALSE));
+        init.put(Part.Type.GREATER_THAN_EQUAL, new Pair<>(CriteriaType.IS_GREATER_THAN, Boolean.FALSE));
+        init.put(Part.Type.LESS_THAN, new Pair<>(CriteriaType.IS_LESS_THAN, Boolean.FALSE));
+        init.put(Part.Type.LESS_THAN_EQUAL, new Pair<>(CriteriaType.IS_LESS_THAN_OR_EQUAL, Boolean.FALSE));
+        init.put(Part.Type.BETWEEN, new Pair<>(CriteriaType.BETWEEN, Boolean.FALSE));
+        init.put(Part.Type.EXISTS, new Pair<>(CriteriaType.EXISTS, Boolean.FALSE));
+        init.put(Part.Type.IS_EMPTY, new Pair<>(CriteriaType.IS_EMPTY, Boolean.FALSE));
+        init.put(Part.Type.IS_NOT_EMPTY, new Pair<>(CriteriaType.IS_EMPTY, Boolean.TRUE));
+        init.put(Part.Type.IS_NULL, new Pair<>(CriteriaType.IS_NULL, Boolean.FALSE));
+        init.put(Part.Type.IS_NOT_NULL, new Pair<>(CriteriaType.IS_NULL, Boolean.TRUE));
+        init.put(Part.Type.CONTAINING, new Pair<>(CriteriaType.CONTAINING, Boolean.FALSE));
+        init.put(Part.Type.NOT_CONTAINING, new Pair<>(CriteriaType.CONTAINING, Boolean.TRUE));
```

Comment:
Part.Type NOT_CONTANING mapping to CriteriaType.CONTAINING ?

Reply:
Agreed...

Have added clarifying comment as below

    // Below table maps the criteria operation indicated by the part of the method name to an actual operation
    // that will be performed by the query.  The Pair<CriteriaType,Boolean> indicates the CriteriaType the Part.Type
    // maps to with the boolean indicating if the logic should be inverted.

Reply:
Hi @Xeinn, I just added a simple test for your added features, and the test failed. Could you grant me developer permission for [forked repo](https://github.com/Xeinn/spring-data-documentdb) to allow me to open a pull request? Thank you.

And by the way, why do you invert the compare, why not match the `IS_GREATER_THAN` as `>`?
 `init.put(CriteriaType.IS_GREATER_THAN,          "r.@?<=@@");`

Reply:
@sophiaso 
Well spotted, let me check, looks initially like that is wrong.

Reply:
@sophiaso I've added you to the repo as a collaborator.

Reply:
Could you give me write permission? Thanks.